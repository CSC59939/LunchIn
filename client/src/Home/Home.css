* {
  user-select: none !important;
}
.home-page {
  display: flex;
  flex-direction: column;
  max-width: 100% !important;
  margin-bottom: 5%;
}
.title-bar {
  padding-top: 5%;
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
  justify-content: center;
}
.home-page-content {
  display: flex;
  flex-direction: column;
  width: 50%;
  margin: 0 auto;
  margin-top: 3%;
  justify-content: center !important;
}
.login-form {
  flex: 1;
  margin: 0 20px;
  padding: 20px !important;
  border-radius: 25px !important;
}
.title-bar .title {
  font-size: 70px;
}
.title-bar .tag-line {
  font-size: 35px;
}
.push-down {
  margin-top: 10px !important;
}
@media only screen and (max-width: 1024px) {
  .title-bar .title {
    font-size: 50px;
  }
  .title-bar .tag-line {
    font-size: 30px;
  }
  .home-page-content {
    width: 65%;
  }
}
@media only screen and (max-width: 650px) {
  .title-bar .title {
    font-size: 40px;
  }
  .title-bar .tag-line {
    font-size: 20px;
  }
  .home-page-content {
    flex-direction: column;
    width: 90%;
  }
}
