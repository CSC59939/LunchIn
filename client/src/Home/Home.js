import React, { Component } from "react";
import "./Home.css";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  IconButton,
  InputAdornment,
  LinearProgress,
  Snackbar,
  TextField,
  Typography,
  Tooltip
} from "@material-ui/core";
import CssBaseline from "@material-ui/core/CssBaseline";
import Visible from "@material-ui/icons/Visibility";
import Hidden from "@material-ui/icons/VisibilityOff";
import { firebase } from "../Config";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
      showPassword: false,
      notify: false,
      notifyMsg: "",
      processing: false,
      message: "Show Password"
    };
    this.handleChange = this.handleChange.bind(this);
    this.login = this.login.bind(this);
  }
  login() {
    this.setState({
      processing: true
    });
    firebase
      .auth()
      .signInWithEmailAndPassword(this.state.email, this.state.password)
      .then(
        resp => {
          this.props.history.push(`/user/home`);
          // wait for AuthStateChange
        },
        err => {
          this.setState({
            notify: true,
            notifyMsg: err.message,
            processing: false
          });
        }
      );
  }
  shouldComponentUpdate(nextProps, nextState) {
    if (nextProps.loggedIn !== this.state.processing) {
      nextState.processing = nextProps.loggedIn;
    }
    return true;
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };
  render() {
    return (
      <div className="home-page">
        <CssBaseline />
        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={3000}
          message={this.state.notifyMsg}
        />

        <div className="title-bar" data-aos="fade-up">
          <Typography
            component="h2"
            variant="h1"
            className="title"
            style={{ color: "#000000" }}
          >
            <strong>LunchIn</strong>
          </Typography>
          <Typography
            component="h2"
            variant="h3"
            className="tag-line"
            data-aos="fade-up"
          >
            Automatically Choose a Lunch Option For You
          </Typography>
        </div>
        <div className="home-page-content">
          <Card className="login-form" data-aos="slide-right">
            {this.state.processing ? (
              <LinearProgress className="loading" />
            ) : null}
            <CardHeader
              className="login-form-header"
              title="Sign In"
              subheader="Your Lunch Option Awaits"
            />
            <CardContent className="login-form-content">
              <TextField
                disabled={this.state.processing}
                className="email"
                id="email"
                label="E-mail"
                value={this.state.email}
                onChange={this.handleChange("email")}
                fullWidth
                required
                type="email"
              />
              <TextField
                disabled={this.state.processing}
                className="password"
                id="password"
                label="Password"
                value={this.state.password}
                onChange={this.handleChange("password")}
                fullWidth
                required
                type={this.state.showPassword ? "text" : "password"}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <Tooltip title={this.state.message} placement="top">
                        <IconButton
                          onClick={() => {
                            if (this.state.message === "Show Password") {
                              this.setState({
                                showPassword: !this.state.showPassword,
                                message: "Hide Password"
                              });
                            } else if (this.state.message === "Hide Password") {
                              this.setState({
                                showPassword: !this.state.showPassword,
                                message: "Show Password"
                              });
                            }
                          }}
                        >
                          {this.state.showPassword ? <Visible /> : <Hidden />}
                        </IconButton>
                      </Tooltip>
                    </InputAdornment>
                  )
                }}
              />
              <Button
                style={{ marginTop: 5, marginBottom: 5 }}
                disabled={this.state.processing}
                variant="contained"
                color="primary"
                fullWidth
                onClick={() => {
                  this.login();
                }}
                className="login"
              >
                Sign-In
              </Button>
              <Typography
                component="h2"
                style={{ marginTop: 5, marginBottom: 5 }}
                variant="subtitle1"
                className="signup-line"
              >
                Don't have an account? Signup now!
              </Typography>
              <Button
                style={{ marginTop: 5, marginBottom: 5 }}
                disabled={this.state.processing}
                variant="contained"
                color="secondary"
                fullWidth
                className="login"
                onClick={() => {
                  this.props.history.push("/sign-up");
                }}
              >
                Sign-Up
              </Button>
            </CardContent>
          </Card>
        </div>
      </div>
    );
  }
}
export default Home;
