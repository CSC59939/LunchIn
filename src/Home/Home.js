import React, { Component } from "react";
import "./Home.css";
import {
  Button,
  Card,
  CardContent,
  CardHeader,
  IconButton,
  InputAdornment,
  LinearProgress,
  Snackbar,
  TextField,
  Typography
} from "@material-ui/core";
import Visible from "@material-ui/icons/Visibility";
import Hidden from "@material-ui/icons/VisibilityOff";
import Next from "@material-ui/icons/CheckCircle";
import Signout from "@material-ui/icons/Cancel";
import { firebase } from "../Config";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      showPassword: false,
      notify: false,
      notifyMsg: "",
      processing: this.props.loggedIn
    };
    this.handleChange = this.handleChange.bind(this);
    this.login = this.login.bind(this);
  }
  login() {
    this.setState({
      processing: true
    });
    firebase
      .auth()
      .signInWithEmailAndPassword(this.state.username, this.state.password)
      .then(
        resp => {
          // wait for AuthStateChange
        },
        err => {
          this.setState({
            notify: true,
            notifyMsg: err.message,
            processing: false
          });
        }
      );
  }
  shouldComponentUpdate(nextProps, nextState) {
    if (nextProps.loggedIn !== this.state.processing) {
      nextState.processing = nextProps.loggedIn;
    }
    return true;
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };
  render() {
    return (
      <div className="home-page">
        <Snackbar
          onClose={() => {
            this.setState({ notify: false, notifyMsg: "" });
          }}
          open={this.state.notify}
          autoHideDuration={3000}
          message={this.state.notifyMsg}
        />

        <div className="title-bar" data-aos="fade-up">
          <Typography variant="display4" className="title">
            <strong>LunchIn</strong>
          </Typography>
          <Typography
            variant="display2"
            className="tag-line"
            data-aos="fade-up"
          >
            Automatically Choose a Lunch Option For You
          </Typography>
        </div>
        <div className="home-page-content">
          <Card className="login-form" data-aos="slide-right">
            {this.state.processing ? (
              <LinearProgress className="loading" />
            ) : null}
            <CardHeader
              className="login-form-header"
              title="Sign In"
              subheader="Your Lunch Option Awaits"
            />
            {this.props.loggedIn === true ? (
              <CardContent className="login-form-content">
                <Typography className="push-down" variant="subheading">
                  Welcome, {firebase.auth().currentUser.displayName}!
                </Typography>
                <Button
                  fullWidth
                  color="primary"
                  variant="raised"
                  className="push-down"
                  onClick={() => {
                    this.props.history.push(`/user/home`);
                  }}
                >
                  <Next style={{ marginRight: "10px" }} />
                  Continue to Your Account
                </Button>
                <Button
                  fullWidth
                  color="secondary"
                  variant="raised"
                  className="push-down"
                  onClick={() => {
                    firebase.auth().signOut();
                  }}
                >
                  <Signout style={{ marginRight: "10px" }} />
                  Sign Out
                </Button>
              </CardContent>
            ) : (
              <CardContent className="login-form-content">
                <TextField
                  disabled={this.state.processing}
                  className="username"
                  id="username"
                  label="E-mail"
                  value={this.state.username}
                  onChange={this.handleChange("username")}
                  fullWidth
                  required
                  type="email"
                />
                <TextField
                  disabled={this.state.processing}
                  className="password"
                  id="password"
                  label="Password"
                  value={this.state.password}
                  onChange={this.handleChange("password")}
                  fullWidth
                  required
                  type={this.state.showPassword ? "text" : "password"}
                  InputProps={{
                    endAdornment: (
                      <InputAdornment position="end">
                        <IconButton
                          onClick={() => {
                            this.setState({
                              showPassword: !this.state.showPassword
                            });
                          }}
                        >
                          {this.state.showPassword ? <Visible /> : <Hidden />}
                        </IconButton>
                      </InputAdornment>
                    )
                  }}
                />
                <Button
                  style={{ marginTop: 5, marginBottom: 5 }}
                  disabled={this.state.processing}
                  variant="raised"
                  color="primary"
                  fullWidth
                  onClick={() => {
                    this.login();
                  }}
                  className="login"
                >
                  Sign-In
                </Button>
                <Typography
                  style={{ marginTop: 5, marginBottom: 5 }}
                  variant="subheading"
                  className="signup-line"
                >
                  Don't have an account? Signup now!
                </Typography>
                <Button
                  style={{ marginTop: 5, marginBottom: 5 }}
                  disabled={this.state.processing}
                  variant="raised"
                  color="secondary"
                  fullWidth
                  className="login"
                  onClick={() => {
                    this.props.history.push("/sign-up");
                  }}
                >
                  Sign-Up
                </Button>
              </CardContent>
            )}
          </Card>
        </div>
      </div>
    );
  }
}
export default Home;
