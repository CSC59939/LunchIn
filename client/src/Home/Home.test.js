import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Home from "./Home";

describe("Home Component", () => {
  it("Should render correctly with correct path", () => {
    const HomeInstance = shallow(<Home />);
    expect(HomeInstance).toMatchSnapshot();
  });
});

describe("When provided with user login info", () => {
  it("Should render correctly", () => {
    const email = "gerry@gmail.com";
    const password = "gerry123";
    const loginInstance = shallow(<Home email={email} password={password} />);
    expect(loginInstance).toMatchSnapshot();
  });
});

describe("When provided with no email for login", () => {
  it("Should render correctly", () => {
    const email = " ";
    const password = "gerry123";
    const loginInstance = shallow(<Home email={email} password={password} />);
    expect(loginInstance).toMatchSnapshot();
  });
});

describe("When provided with incorrect credential for login", () => {
  it("Should render correctly", () => {
    const email = "ajfakljfad@test.com";
    const password = "kajasjdalskj";
    const loginInstance = shallow(<Home email={email} password={password} />);
    expect(loginInstance).toMatchSnapshot();
  });
});

describe("When provided with no email or password for login", () => {
  it("Should render correctly", () => {
    const email = "";
    const password = "";
    const loginInstance = shallow(<Home email={email} password={password} />);
    expect(loginInstance).toMatchSnapshot();
  });
});

describe("When provided with no password for login", () => {
  it("Should render correctly", () => {
    const email = "gerry@gmail.com";
    const password = "";
    const loginInstance = shallow(<Home email={email} password={password} />);
    expect(loginInstance).toMatchSnapshot();
  });
});
