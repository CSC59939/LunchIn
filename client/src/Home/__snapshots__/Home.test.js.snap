// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should render correctly with correct path 1`] = `
<div
  className="home-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={3000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="title-bar"
    data-aos="fade-up"
  >
    <WithStyles(Typography)
      className="title"
      component="h2"
      style={
        Object {
          "color": "#000000",
        }
      }
      variant="h1"
    >
      <strong>
        LunchIn
      </strong>
    </WithStyles(Typography)>
    <WithStyles(Typography)
      className="tag-line"
      component="h2"
      data-aos="fade-up"
      variant="h3"
    >
      Automatically Choose a Lunch Option For You
    </WithStyles(Typography)>
  </div>
  <div
    className="home-page-content"
  >
    <WithStyles(Card)
      className="login-form"
      data-aos="slide-right"
    >
      <WithStyles(CardHeader)
        className="login-form-header"
        subheader="Your Lunch Option Awaits"
        title="Sign In"
      />
      <WithStyles(CardContent)
        className="login-form-content"
      >
        <form>
          <TextField
            className="email"
            disabled={false}
            fullWidth={true}
            id="email"
            label="E-mail"
            onChange={[Function]}
            required={true}
            select={false}
            type="email"
            value=""
            variant="standard"
          />
          <TextField
            InputProps={
              Object {
                "endAdornment": <WithStyles(InputAdornment)
                  position="end"
                >
                  <WithStyles(Tooltip)
                    placement="top"
                    title="Show Password"
                  >
                    <WithStyles(IconButton)
                      onClick={[Function]}
                    >
                      <pure(VisibilityOffIcon) />
                    </WithStyles(IconButton)>
                  </WithStyles(Tooltip)>
                </WithStyles(InputAdornment)>,
              }
            }
            className="password"
            disabled={false}
            fullWidth={true}
            id="password"
            label="Password"
            onChange={[Function]}
            required={true}
            select={false}
            type="password"
            value=""
            variant="standard"
          />
          <WithStyles(Button)
            className="login"
            color="primary"
            disabled={false}
            fullWidth={true}
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              }
            }
            variant="contained"
          >
            Sign-In
          </WithStyles(Button)>
        </form>
        <WithStyles(Typography)
          className="signup-line"
          component="h2"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="subtitle1"
        >
          Don't have an account? Signup now!
        </WithStyles(Typography)>
        <WithStyles(Button)
          className="login"
          color="secondary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="contained"
        >
          Sign-Up
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;

exports[`Home Component When provided with incorrect credential for login Should render correctly 1`] = `
<div
  className="home-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={3000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="title-bar"
    data-aos="fade-up"
  >
    <WithStyles(Typography)
      className="title"
      component="h2"
      style={
        Object {
          "color": "#000000",
        }
      }
      variant="h1"
    >
      <strong>
        LunchIn
      </strong>
    </WithStyles(Typography)>
    <WithStyles(Typography)
      className="tag-line"
      component="h2"
      data-aos="fade-up"
      variant="h3"
    >
      Automatically Choose a Lunch Option For You
    </WithStyles(Typography)>
  </div>
  <div
    className="home-page-content"
  >
    <WithStyles(Card)
      className="login-form"
      data-aos="slide-right"
    >
      <WithStyles(CardHeader)
        className="login-form-header"
        subheader="Your Lunch Option Awaits"
        title="Sign In"
      />
      <WithStyles(CardContent)
        className="login-form-content"
      >
        <form>
          <TextField
            className="email"
            disabled={false}
            fullWidth={true}
            id="email"
            label="E-mail"
            onChange={[Function]}
            required={true}
            select={false}
            type="email"
            value=""
            variant="standard"
          />
          <TextField
            InputProps={
              Object {
                "endAdornment": <WithStyles(InputAdornment)
                  position="end"
                >
                  <WithStyles(Tooltip)
                    placement="top"
                    title="Show Password"
                  >
                    <WithStyles(IconButton)
                      onClick={[Function]}
                    >
                      <pure(VisibilityOffIcon) />
                    </WithStyles(IconButton)>
                  </WithStyles(Tooltip)>
                </WithStyles(InputAdornment)>,
              }
            }
            className="password"
            disabled={false}
            fullWidth={true}
            id="password"
            label="Password"
            onChange={[Function]}
            required={true}
            select={false}
            type="password"
            value=""
            variant="standard"
          />
          <WithStyles(Button)
            className="login"
            color="primary"
            disabled={false}
            fullWidth={true}
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              }
            }
            variant="contained"
          >
            Sign-In
          </WithStyles(Button)>
        </form>
        <WithStyles(Typography)
          className="signup-line"
          component="h2"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="subtitle1"
        >
          Don't have an account? Signup now!
        </WithStyles(Typography)>
        <WithStyles(Button)
          className="login"
          color="secondary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="contained"
        >
          Sign-Up
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;

exports[`Home Component When provided with no email for login Should render correctly 1`] = `
<div
  className="home-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={3000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="title-bar"
    data-aos="fade-up"
  >
    <WithStyles(Typography)
      className="title"
      component="h2"
      style={
        Object {
          "color": "#000000",
        }
      }
      variant="h1"
    >
      <strong>
        LunchIn
      </strong>
    </WithStyles(Typography)>
    <WithStyles(Typography)
      className="tag-line"
      component="h2"
      data-aos="fade-up"
      variant="h3"
    >
      Automatically Choose a Lunch Option For You
    </WithStyles(Typography)>
  </div>
  <div
    className="home-page-content"
  >
    <WithStyles(Card)
      className="login-form"
      data-aos="slide-right"
    >
      <WithStyles(CardHeader)
        className="login-form-header"
        subheader="Your Lunch Option Awaits"
        title="Sign In"
      />
      <WithStyles(CardContent)
        className="login-form-content"
      >
        <form>
          <TextField
            className="email"
            disabled={false}
            fullWidth={true}
            id="email"
            label="E-mail"
            onChange={[Function]}
            required={true}
            select={false}
            type="email"
            value=""
            variant="standard"
          />
          <TextField
            InputProps={
              Object {
                "endAdornment": <WithStyles(InputAdornment)
                  position="end"
                >
                  <WithStyles(Tooltip)
                    placement="top"
                    title="Show Password"
                  >
                    <WithStyles(IconButton)
                      onClick={[Function]}
                    >
                      <pure(VisibilityOffIcon) />
                    </WithStyles(IconButton)>
                  </WithStyles(Tooltip)>
                </WithStyles(InputAdornment)>,
              }
            }
            className="password"
            disabled={false}
            fullWidth={true}
            id="password"
            label="Password"
            onChange={[Function]}
            required={true}
            select={false}
            type="password"
            value=""
            variant="standard"
          />
          <WithStyles(Button)
            className="login"
            color="primary"
            disabled={false}
            fullWidth={true}
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              }
            }
            variant="contained"
          >
            Sign-In
          </WithStyles(Button)>
        </form>
        <WithStyles(Typography)
          className="signup-line"
          component="h2"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="subtitle1"
        >
          Don't have an account? Signup now!
        </WithStyles(Typography)>
        <WithStyles(Button)
          className="login"
          color="secondary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="contained"
        >
          Sign-Up
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;

exports[`Home Component When provided with no email or password for login Should render correctly 1`] = `
<div
  className="home-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={3000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="title-bar"
    data-aos="fade-up"
  >
    <WithStyles(Typography)
      className="title"
      component="h2"
      style={
        Object {
          "color": "#000000",
        }
      }
      variant="h1"
    >
      <strong>
        LunchIn
      </strong>
    </WithStyles(Typography)>
    <WithStyles(Typography)
      className="tag-line"
      component="h2"
      data-aos="fade-up"
      variant="h3"
    >
      Automatically Choose a Lunch Option For You
    </WithStyles(Typography)>
  </div>
  <div
    className="home-page-content"
  >
    <WithStyles(Card)
      className="login-form"
      data-aos="slide-right"
    >
      <WithStyles(CardHeader)
        className="login-form-header"
        subheader="Your Lunch Option Awaits"
        title="Sign In"
      />
      <WithStyles(CardContent)
        className="login-form-content"
      >
        <form>
          <TextField
            className="email"
            disabled={false}
            fullWidth={true}
            id="email"
            label="E-mail"
            onChange={[Function]}
            required={true}
            select={false}
            type="email"
            value=""
            variant="standard"
          />
          <TextField
            InputProps={
              Object {
                "endAdornment": <WithStyles(InputAdornment)
                  position="end"
                >
                  <WithStyles(Tooltip)
                    placement="top"
                    title="Show Password"
                  >
                    <WithStyles(IconButton)
                      onClick={[Function]}
                    >
                      <pure(VisibilityOffIcon) />
                    </WithStyles(IconButton)>
                  </WithStyles(Tooltip)>
                </WithStyles(InputAdornment)>,
              }
            }
            className="password"
            disabled={false}
            fullWidth={true}
            id="password"
            label="Password"
            onChange={[Function]}
            required={true}
            select={false}
            type="password"
            value=""
            variant="standard"
          />
          <WithStyles(Button)
            className="login"
            color="primary"
            disabled={false}
            fullWidth={true}
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              }
            }
            variant="contained"
          >
            Sign-In
          </WithStyles(Button)>
        </form>
        <WithStyles(Typography)
          className="signup-line"
          component="h2"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="subtitle1"
        >
          Don't have an account? Signup now!
        </WithStyles(Typography)>
        <WithStyles(Button)
          className="login"
          color="secondary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="contained"
        >
          Sign-Up
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;

exports[`Home Component When provided with no password for login Should render correctly 1`] = `
<div
  className="home-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={3000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="title-bar"
    data-aos="fade-up"
  >
    <WithStyles(Typography)
      className="title"
      component="h2"
      style={
        Object {
          "color": "#000000",
        }
      }
      variant="h1"
    >
      <strong>
        LunchIn
      </strong>
    </WithStyles(Typography)>
    <WithStyles(Typography)
      className="tag-line"
      component="h2"
      data-aos="fade-up"
      variant="h3"
    >
      Automatically Choose a Lunch Option For You
    </WithStyles(Typography)>
  </div>
  <div
    className="home-page-content"
  >
    <WithStyles(Card)
      className="login-form"
      data-aos="slide-right"
    >
      <WithStyles(CardHeader)
        className="login-form-header"
        subheader="Your Lunch Option Awaits"
        title="Sign In"
      />
      <WithStyles(CardContent)
        className="login-form-content"
      >
        <form>
          <TextField
            className="email"
            disabled={false}
            fullWidth={true}
            id="email"
            label="E-mail"
            onChange={[Function]}
            required={true}
            select={false}
            type="email"
            value=""
            variant="standard"
          />
          <TextField
            InputProps={
              Object {
                "endAdornment": <WithStyles(InputAdornment)
                  position="end"
                >
                  <WithStyles(Tooltip)
                    placement="top"
                    title="Show Password"
                  >
                    <WithStyles(IconButton)
                      onClick={[Function]}
                    >
                      <pure(VisibilityOffIcon) />
                    </WithStyles(IconButton)>
                  </WithStyles(Tooltip)>
                </WithStyles(InputAdornment)>,
              }
            }
            className="password"
            disabled={false}
            fullWidth={true}
            id="password"
            label="Password"
            onChange={[Function]}
            required={true}
            select={false}
            type="password"
            value=""
            variant="standard"
          />
          <WithStyles(Button)
            className="login"
            color="primary"
            disabled={false}
            fullWidth={true}
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              }
            }
            variant="contained"
          >
            Sign-In
          </WithStyles(Button)>
        </form>
        <WithStyles(Typography)
          className="signup-line"
          component="h2"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="subtitle1"
        >
          Don't have an account? Signup now!
        </WithStyles(Typography)>
        <WithStyles(Button)
          className="login"
          color="secondary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="contained"
        >
          Sign-Up
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;

exports[`Home Component When provided with user login info Should render correctly 1`] = `
<div
  className="home-page"
>
  <WithStyles(CssBaseline) />
  <WithStyles(Snackbar)
    autoHideDuration={3000}
    message=""
    onClose={[Function]}
    open={false}
  />
  <div
    className="title-bar"
    data-aos="fade-up"
  >
    <WithStyles(Typography)
      className="title"
      component="h2"
      style={
        Object {
          "color": "#000000",
        }
      }
      variant="h1"
    >
      <strong>
        LunchIn
      </strong>
    </WithStyles(Typography)>
    <WithStyles(Typography)
      className="tag-line"
      component="h2"
      data-aos="fade-up"
      variant="h3"
    >
      Automatically Choose a Lunch Option For You
    </WithStyles(Typography)>
  </div>
  <div
    className="home-page-content"
  >
    <WithStyles(Card)
      className="login-form"
      data-aos="slide-right"
    >
      <WithStyles(CardHeader)
        className="login-form-header"
        subheader="Your Lunch Option Awaits"
        title="Sign In"
      />
      <WithStyles(CardContent)
        className="login-form-content"
      >
        <form>
          <TextField
            className="email"
            disabled={false}
            fullWidth={true}
            id="email"
            label="E-mail"
            onChange={[Function]}
            required={true}
            select={false}
            type="email"
            value=""
            variant="standard"
          />
          <TextField
            InputProps={
              Object {
                "endAdornment": <WithStyles(InputAdornment)
                  position="end"
                >
                  <WithStyles(Tooltip)
                    placement="top"
                    title="Show Password"
                  >
                    <WithStyles(IconButton)
                      onClick={[Function]}
                    >
                      <pure(VisibilityOffIcon) />
                    </WithStyles(IconButton)>
                  </WithStyles(Tooltip)>
                </WithStyles(InputAdornment)>,
              }
            }
            className="password"
            disabled={false}
            fullWidth={true}
            id="password"
            label="Password"
            onChange={[Function]}
            required={true}
            select={false}
            type="password"
            value=""
            variant="standard"
          />
          <WithStyles(Button)
            className="login"
            color="primary"
            disabled={false}
            fullWidth={true}
            onClick={[Function]}
            style={
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              }
            }
            variant="contained"
          >
            Sign-In
          </WithStyles(Button)>
        </form>
        <WithStyles(Typography)
          className="signup-line"
          component="h2"
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="subtitle1"
        >
          Don't have an account? Signup now!
        </WithStyles(Typography)>
        <WithStyles(Button)
          className="login"
          color="secondary"
          disabled={false}
          fullWidth={true}
          onClick={[Function]}
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
            }
          }
          variant="contained"
        >
          Sign-Up
        </WithStyles(Button)>
      </WithStyles(CardContent)>
    </WithStyles(Card)>
  </div>
</div>
`;
